### ISSUE TITLE
Reference incorrectly created for `this` within a JSXIdentifier

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#4055]
**Repro**

```TS
import React from 'react';

class Hello extends React.Component<{tag: () => React$MixedElement}> {
  inline() {
    return <this.props.tag />
  }
}
```

**Expected Result**

The `this` should be treated like a `ThisExpression`.

**Actual Result**

The `this` is treated like a variable called `this`, and thus creates an unclosed reference to a variable that can never exist.

**Additional Info**

Context: https://github.com/typescript-eslint/typescript-eslint/issues/3788#issuecomment-951081315
cc @ljharb

**Comment by juank1809** [comment]
Hi, if this issue is still accepting PR's, I'd glad to help

**Comment by bradzacher** [comment]
Feel free @juank1809!

**Comment by juank1809** [comment]
So, after days thinking on how to solve this problem I got stuck! I do not know if I'm heading into the right way. So I was wondering if you guys could give me some guidance about it😅.   
This is what I've made so far: https://github.com/juank1809/typescript-eslint/commit/3d607673c98132315b9b6fd806d60013ae3b737d.

Thank you in advance!

**Comment by bradzacher** [comment]
[Here is a link where you can see the AST](https://typescript-eslint.io/play/#ts=4.5.2&jsx=true&sourceType=module&showAST=es&code=CYUwxgNghgTiAEYD2A7AzgF3gJRFMGA3AFDGRRprwBiSS8A3sfC-AGZ0Bc8UKAniVbwAtiAwALJMAAUASkbMhraQB4JASzQA6DvQD0APlmDWAX2KmgA&rules=N4XyA&tsConfig=N4XyA)

You'll see the important bit will look a little like this:
```
{
  type: 'JSXMemberExpression',
  object: {
    type: 'JSXIdentifier',
    name: 'this',
  },
  property: { ... },
}
```

We want to ignore the the `JSXIdentifier` in this specific case - when its `.name` is `this` AND it is the `.object` of a `JSXMemberExpression`.
You won't need to change the AST types at all for this - just add some logic to the `scope-manager`!
